### PR TITLE
Remove dependency on fluent/mixin/config_placeholders

### PR DESCRIPTION
**Pull request by t0ffel**
Hostname placeholder `__HOSTNAME__` or `${hostname}` will be expanded to `#{Socket.gethostname}` or to value specified in `hostname` field. 

fixes https://github.com/tagomoris/fluent-plugin-secure-forward/issues/38
Please let me know what you think.

/a bit of offtopic/
When I run with trace on I get a bunch of errors. Although data seem to be getting in. Is that a problem?
```
2016-06-09 13:36:02 -0400 [trace]: plugin/output_node.rb:347:rescue in block in connect: SSLError error_class=OpenSSL::SSL::SSLErrorWaitReadable error=#<OpenSSL::SSL::SSLErrorWaitReadable: read would block> mtime=2016-06-09 13:35:55 -0400 host="10.3.14.104" port=24284
2016-06-09 13:36:02 -0400 [trace]: plugin/output_node.rb:347:rescue in block in connect: SSLError error_class=OpenSSL::SSL::SSLErrorWaitReadable error=#<OpenSSL::SSL::SSLErrorWaitReadable: read would block> mtime=2016-06-09 13:35:55 -0400 host="10.3.14.104" port=24284
```

### CODE REVIEW

**Comment by t0ffel** [comment]
@tagomoris can you please take a look?

**Comment by tagomoris** [comment]
Sorry to miss this.
Almost code looks good to me, but it's needed to add tests about hostname placeholders.
I'll add some comments about some lines of code.

Trace logs about SSLError are normal for this plugin.

File: lib/fluent/plugin/in_secure_forward.rb [hunk 1]
```diff
@@ -91,9 +90,30 @@ def initialize
       define_method("router") { Fluent::Engine }
     end
 
+    def replace_hostname_placeholder(conf)
+      hostname = @hostname ? @hostname : Socket.gethostname
+
+      placeholders = ['__HOSTNAME__', '${hostname}']
+
+
```
Comment (tagomoris) [inline lib/fluent/plugin/in_secure_forward.rb]: Why is there 2 lines whitespace?

File: lib/fluent/plugin/in_secure_forward.rb [hunk 2]
```diff
@@ -91,9 +90,30 @@ def initialize
...
+      check_element = ->(placeholders, c) {
+        c.keys.each do |k|
+          v = c.fetch(k, nil)
```
Comment (tagomoris) [inline lib/fluent/plugin/in_secure_forward.rb]: `v = c[k]` is enough. `c.fetch(k, nil)` is same with it.

File: lib/fluent/plugin/in_secure_forward.rb [hunk 3]
```diff
@@ -91,9 +90,30 @@ def initialize
...
+          if v and placeholders.include? v
```
Comment (tagomoris) [inline lib/fluent/plugin/in_secure_forward.rb]: `placeholders.include? v` works only when `v == "${hostname}"` or `v == "__HOSTNAME__"`. But it should work when string values includes `${hostname}` or `__HOSTNAME__` (and replace these placeholders with hostname string).

Comment (tagomoris) [inline lib/fluent/plugin/in_secure_forward.rb]: This code doesn't require high performance, so we can write it like below:
```ruby
c.keys.each do |k|
  if v = c[k] && placeholders.any?{|ph| v.include?(ph) }
    c[k] = v.gsub('${hostname}', hostname).gsub('__HOSTNAME__', hostname)
  end
end
# ...
```

**Comment by tagomoris** [comment]
I merged #44 instead of merging this pull-request.
Thank you for contribution!

**Pull request closed without merging**